src/scaffoldmaker/meshtypes/meshtype_3d_wholebody2.py

```diff
@@ -1,199 +1,675 @@
+        options["Torso width"] = 3.2
+        options["Pelvis drop"] = 1.5
+        options["Pelvis width"] = 2.0
+        options["Leg lateral angle degrees"] = 10.0
+        options["Leg length"] = 10.0
+        options["Leg top diameter"] = 2.0
+        options["Leg bottom diameter"] = 0.7
+        options["Foot height"] = 1.0
+        options["Foot length"] = 2.5
+        options["Foot thickness"] = 0.3
+        options["Foot width"] = 1.0
+        options["Inner proportion default"] = 0.7
+        options["Inner proportion head"] = 0.35
+        return options
+
+    @classmethod
+    def getOrderedOptionNames(cls):
+        return [
+            "Head depth",
+            "Head length",
+            "Head width",
+            "Neck length",
+            "Shoulder drop",
+            "Shoulder width",
+            "Arm lateral angle degrees",
+            "Arm length",
+            "Arm top diameter",
+            "Wrist thickness",
+            "Wrist width",
+            "Hand length",
+            "Hand thickness",
+            "Hand width",
+            "Thorax length",
+            "Abdomen length",
+            "Torso depth",
+            "Torso width",
+            "Pelvis drop",
+            "Pelvis width",
+            "Leg lateral angle degrees",
+            "Leg length",
+            "Leg top diameter",
+            "Leg bottom diameter",
+            "Foot height",
+            "Foot length",
+            "Foot thickness",
+            "Foot width",
+            "Inner proportion default",
+            "Inner proportion head"
+        ]
+
+    @classmethod
+    def checkOptions(cls, options):
+        dependentChanges = False
+        for key in [
+            "Head depth",
+            "Head length",
+            "Head width",
+            "Neck length",
+            "Shoulder drop",
+            "Shoulder width",
+            "Arm length",
+            "Arm top diameter",
+            "Wrist thickness",
+            "Wrist width",
+            "Hand length",
+            "Hand thickness",
+            "Hand width",
+            "Pelvis drop",
+            "Pelvis width",
+            "Thorax length",
+            "Abdomen length",
+            "Torso depth",
+            "Torso width",
+            "Leg length",
+            "Leg top diameter",
+            "Leg bottom diameter",
+            "Foot height",
+            "Foot length",
+            "Foot thickness",
+            "Foot width"
+        ]:
+            if options[key] < 0.1:
+                options[key] = 0.1
+        for key in [
+            "Inner proportion default",
+            "Inner proportion head"
+        ]:
+            if options[key] < 0.1:
+                options[key] = 0.1
+            elif options[key] > 0.9:
+                options[key] = 0.9
+        for key in [
+            "Arm lateral angle degrees"
+        ]:
+            if options[key] < -60.0:
+                options[key] = -60.0
+            elif options[key] > 200.0:
```

Comment:
I wonder if 200 degrees is the appropriate limit. It fails to generate properly at 200 deg anyway and I think the arms' realistic lateral limit is closer to around 180 degs.

Reply:
The parameter bounds do not guarantee correctness, only not crashing & I'd like to be able to experiment when I next want to look into it.